### PR TITLE
Fix XLSX column names for remapped cols

### LINKED ISSUE #18573: Remapped columns do not use renamed titles in XLSX exports
**Describe the bug**
If a column uses remapped values, and is renamed, an XLSX export will not use the new name 

**To Reproduce**
1. In the data model settings, change a column to use remapped values
2. In a question using a table visualization, rename the remapped column
3. Export as XLSX

**Expected behavior**
Column name should match what is shown in Metabase exactly

**Information about your Metabase Installation:**
Current master

### PR DESCRIPTION
**Pull request by noahmoss**
If a column is remapped to a FK field, its `:id` in the col metadata will be the FK id. But the display title in the viz settings will still be associated with the original field id. So we need to pull out the original field ID from the `:fk_field_id` field, if it exists, and use it to look up the correct column in the viz settings.

Fixes #18573

### CODE REVIEW

**Comment by noahmoss** [comment]
Hey @nemanjaglumac  --

The repro you created for #18573 tries both CSV and XLSX exports, even though the specific issue fixed here is only relevant for XLSX. I've removed the CSV export from the repro so that I can unskip it in this PR. 

#18572 is the general issue for renamed columns in CSV & JSON, which we don't support at all right now. Perhaps a separate repro should be made for that issue for those file types?

**Review by howonlee — APPROVED** [review]
Nit fix works

**Review by metamben — COMMENTED** [review]
If `:fk_field_id` is only set when a col is remapped to a foreign key field, then this good, I think.

**Comment by noahmoss** [comment]
@metamben Good catch, it's also set during implicit joins. Using `:fk_field_id` iff `:remapped_from` is also set seems to do the trick.

File: src/metabase/query_processor/streaming/xlsx.clj [hunk 1]
```diff
@@ -11,11 +11,11 @@
...
-  (:import java.io.OutputStream
-           [java.time LocalDate LocalDateTime LocalTime OffsetDateTime OffsetTime ZonedDateTime]
-           [org.apache.poi.ss.usermodel Cell DataFormat DateUtil Workbook]
-           org.apache.poi.ss.util.CellRangeAddress
-           [org.apache.poi.xssf.streaming SXSSFRow SXSSFSheet SXSSFWorkbook]))
+  (:import (java.io OutputStream)
+           (java.time LocalDate LocalDateTime LocalTime OffsetDateTime OffsetTime ZonedDateTime)
+           (org.apache.poi.ss.usermodel Cell DataFormat DateUtil Workbook)
+           (org.apache.poi.ss.util CellRangeAddress)
+           (org.apache.poi.xssf.streaming SXSSFRow SXSSFSheet SXSSFWorkbook)))
```
Comment (metamben) [inline src/metabase/query_processor/streaming/xlsx.clj:18]: Is this the new standard `:import` syntax? I had the same and the namespace checker made me create one like you are changing here.

**Review by metamben — APPROVED** [review]
Apart from the import syntax LGTM.

File: src/metabase/query_processor/streaming/xlsx.clj (hunk 1 above)
Comment (noahmoss) [reply src/metabase/query_processor/streaming/xlsx.clj:18]: Oh, I think my editor plugin makes this change automatically when I run "clean namespaces". I'm not sure what the standard is. I'll revert this manually.

Comment (noahmoss) [reply src/metabase/query_processor/streaming/xlsx.clj]: I don't know if we really have a preference...may just be a quirk in the linter. Might be a good thing to bring up on Slack or at a backend sync 👍

**Pull request merged**